### PR TITLE
chore: Stricter ESLint config

### PR DESCRIPTION
**Pull request by going-confetti**
Relevant issue: https://github.com/grafana/k6-studio/issues/277

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -43,11 +47,8 @@ module.exports = {
         argsIgnorePattern: '^_',
       },
     ],
-    // TODO: remove warnings when plugin:@typescript-eslint/recommended-type-checked is enabled
-    '@typescript-eslint/no-unsafe-assignment': 'warn',
-    '@typescript-eslint/no-unsafe-return': 'warn',
-    '@typescript-eslint/no-unsafe-argument': 'warn',
-    '@typescript-eslint/no-unsafe-member-access': 'warn',
+    '@typescript-eslint/no-floating-promises': 'off',
+    '@typescript-eslint/no-misused-promises': 'off',
```
Comment (going-confetti) [inline .eslintrc.cjs:51]: I suggest turning those on later. It highlighted some cases where we pass async functions as e.g. click handlers

File: src/codegen/codegen.utils.ts [hunk 2]
```diff
@@ -24,6 +24,8 @@ export function stringify(value: unknown): string {
     return `{${properties}}`
   }
 
+  // TODO: https://github.com/grafana/k6-studio/issues/277
```
Comment (going-confetti) [inline src/codegen/codegen.utils.ts:27]: Marked ignore comments with a link to the issue (I'll update the issue description once this PR is merged).
While it does look annoying, the pros of enabling stricter rules for everything else outweigh the cons

File: src/views/Validator/Validator.hooks.ts [hunk 3]
```diff
@@ -2,7 +2,8 @@ import { useLocation, useParams } from 'react-router-dom'
 
 export function useScriptPath() {
   const { fileName } = useParams()
-  const { state } = useLocation()
+  // TODO(router): useLocation is not type-safe. Refactor this route to avoid using it.
+  const { state } = useLocation() as { state: { externalScriptPath: string } }
```
Comment (going-confetti) [inline src/views/Validator/Validator.hooks.ts:6]: @allansson now I see what you meant, and I barely scratched the surface...

Comment (allansson) [reply src/views/Validator/Validator.hooks.ts:6]: The same goes for getting route params via `useParams` or search params via `useSearchParams`. You're just casting things hoping that your routes are structured in a certain way.

**Review by e-fisher — DISMISSED** [review]

File: .eslintrc.cjs (hunk 1 above)
Comment (going-confetti) [reply .eslintrc.cjs]: Initially, I was reluctant to do this, as the number of warning seemed to be too high with quite a few of them not being useful. I then discussed possible solutions with @e-fisher and we found a way to enable these rules with a few tweaks.

**Pull request merged**